apps/google/sheets/services.py

```
@@ -0,0 +1,78 @@
+                "spreadsheet_id": sheet.spreadsheet_id,
+            },
+        )
+
+    def _load_public(self, sheet: GoogleSheet) -> SheetLoadResult:
+        csv_url = f"https://docs.google.com/spreadsheets/d/{sheet.spreadsheet_id}/gviz/tq?tqx=out:csv"
+        response = requests.get(csv_url, timeout=self.timeout)
+        response.raise_for_status()
+        first_line = response.text.splitlines()[0] if response.text else ""
+        headers = [part.strip().strip('"') for part in first_line.split(",") if part.strip()]
+        if not headers:
+            raise ValidationError("Could not detect headers in the public sheet.")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)